kernel/src/row_tracking.rs

```diff
@@ -105,14 +100,19 @@ impl RowTrackingVisitor {
 
 impl RowVisitor for RowTrackingVisitor {
     fn selected_column_names_and_types(&self) -> (&'static [ColumnName], &'static [DataType]) {
-        static NAMES_AND_TYPES: LazyLock<ColumnNamesAndTypes> =
-            LazyLock::new(|| add_files_schema().leaves(None));
+        static NAMES_AND_TYPES: LazyLock<ColumnNamesAndTypes> = LazyLock::new(|| {
```

Comment:
this change is just an optimization since we only care about this one field?

Reply:
Yes, it is an optimization but more importantly it decouples this logic from add_files_schema()